django_bmemcached/memcached.py

```diff
@@ -38,12 +68,13 @@ def _cache(self):
             return client
 
         if self._options:
-            client = self._lib.Client(
-                self._servers, self._options.get('username', None),
-                self._options.get('password', None)
-            )
+            client = self._lib.Client(self._servers, **self._options)
+            try:
+                client = self._lib.Client(self._servers, **self._options)
+            except TypeError:
+                raise InvalidCacheOptions(self._options)
```

Comment:
Instead of checking options each time, this will raise an error on init. Given that this is a config / settings issue - it will only occur when settings are invalid, and once they are updated it will never fail again, so the overhead is effectively zero.